client/object_replicate.go

```diff
@@ -198,12 +207,17 @@ func newReplicateMessage(id oid.ID, src io.ReadSeeker, signer neofscrypto.Signer
 	msg = protowire.AppendBytes(msg, idSig)
 	msg = protowire.AppendTag(msg, fieldNumSigScheme, protowire.VarintType)
 	msg = protowire.AppendVarint(msg, sigScheme)
+	msg = protowire.AppendTag(msg, fieldNumSignObjectFlag, protowire.VarintType)
+	msg = protowire.AppendVarint(msg, protowire.EncodeBool(requireObjectSignature))
 
 	return msg, nil
 }
 
 type replicateResponse struct {
-	err error
+	_sigRequested bool
```

Comment:
C-style?

Reply:
Looks weird. And it's not a response field, really.

Reply:
this is the only possibility of understanding a signature is missing but was requested in the current code. we also may just do nothing if it is missing and return nil, it is just an RPC wrapper without any logic then, maybe that is good too, not sure

> And it's not a response field, really

yes, that is why it is underscored...

> Looks weird

... which is available by the language and not prohibited by any of our liners